cli/cli-improvements.md

```diff
@@ -0,0 +1,315 @@
+# CLI improvements
+
+* **Status**: Draft
+* **Author**: Nithya Subramanian (@nithyatsu)
+
+## Overview
+
+When a Radius user wants to deploy an application, a few concepts come into picture
+
+*	Resource group – Radius resource group mimic ARM resource group and are tied to RBAC. 
+*	Environment – Radius environment is what the application uses. The environment can have a set of recipes and is “in” a resource group itself, since it requires RBAC to make sure not everyone can edit it and for example, point a Production environment to use Sandbox recipes.
+*	Workspace – Radius workspace has a "per user" information about the kubernetes context/ cluster where the Radius services are hosted as well as the “default” environment and resource group  of the user.
+
+There are a few points that can be improved in the way these concepts work together:
+
+*	Today, when we rad init,  we make a “default” resource group which creates a single resource group perception for radius users. However, we would want multiple resource groups for multiple copies of same application, based on the scenario. For instance, a cool-app deployment to production and the same cool-app deployment to staging would not need the same RBAC and hence have to be two copies in two resource groups. Also, they have to be in different resource groups so that there are no resource-id conflicts. Would it be a better idea, to not create default group during rad init, and create/use a resource group based on what the users chooses as the application name and its environment [production, staging,[enter your own]]?
+
+
+*	Today, if we want two copies of same app, we want to create two resource group, have an environment in each of those resource group, have the user explicitly use rad group switch and rad env switch or specify the group and env using flags, to make a desirable deployment. Would it be a better idea to, instead of expecting users specify instructions explicitly, print a default behavior which is intuitive and deploy the app if user confirms?
+
+
+*	Since only ResourceGroup  is part of resource ID, if we deploy multiple application to same resource group (ex: we do this in our functional tests) we have to ensure the resource names across applications are all unique. This limitation has no reason to be present in real world and is a hinderance since the developer has to make sure to choose names that are not already taken across applications. Again, we could create a resource group based on user scenario per instance of application using an agreed upon convention to solve this problem. 
+
```

Comment:
I think this is a pretty good intro. I'm expecting @willtsai to provide this as part of the feature spec, and you can reference it here. 

These are good examples.